client/network/src/protocol/sync.rs

```diff
@@ -620,6 +687,17 @@ impl<B: BlockT> ChainSync<B> {
 					return Ok(None)
 				}
 
+				if let SyncMode::Warp = &self.mode {
+					// TODO: wait for 3 peers
+					if self.warp_sync.is_none() {
+						log::debug!(target: "sync", "Starting warp state sync.");
+						if let Some(provider) = &self.warp_sync_provider {
+							self.warp_sync =
+								Some(WarpSync::new(self.client.clone(), provider.clone()));
```

Comment:
Does that mean we will start again from zero when the node is restarted?

Reply:
Yes. Proofs and state are downloaded into memory anyway. Block history download won't be restarted however, but that's coming in the following PR.